### PR TITLE
Resolve disk name collisions for attached regional disks

### PR DESCRIPTION
**Pull request by davidz627**
Fixes: #109 

/assign @msau42 
/assign @verult

### CODE REVIEW

**Comment by davidz627** [comment]
ping @msau42 @verult PTAL when you have some time

File: pkg/gce-pd-csi-driver/node.go [hunk 1]
```diff
@@ -177,7 +177,12 @@ func (ns *GCENodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStage
...
 	partition := ""
 
-	devicePaths := ns.DeviceUtils.GetDiskByIdPaths(volumeKey.Name, partition)
+	deviceName, err := common.GetDeviceName(volumeKey)
+	if err != nil {
+		status.Error(codes.Internal, fmt.Sprintf("error getting device name: %v", err))
+	}
+
+	devicePaths := ns.DeviceUtils.GetDiskByIdPaths(deviceName, partition)
 	devicePath, err := ns.DeviceUtils.VerifyDevicePath(devicePaths)
```
Comment (msau42) [inline pkg/gce-pd-csi-driver/node.go]: did you merge that udev fix that used scsi_id and validated the device name?  I wonder if that will have issues with this change.

Comment (davidz627) [reply pkg/gce-pd-csi-driver/node.go]: hmm I'm not really sure what you're talking about. I have a vague memory of something like that but I can't find a PR about it...

Comment (davidz627) [reply pkg/gce-pd-csi-driver/node.go]: do you mean the change you made to k8s/k8s where you try to rerun `udevadm trigger` a bunch of times after doing some file parsing for validation? If so, then no I haven't merged that

Comment (msau42) [reply pkg/gce-pd-csi-driver/node.go]: ok we may have to revisit this in that PR then

**Pull request merged**